core/src/test/java/se/tre/freki/labels/LabelClientTypeContextTest.java

```diff
@@ -152,4 +155,45 @@ public void createIdPublishesEventOnSuccess() throws Exception {
     typeContext.createId("foo").get();
     verify(eventBus).post(any(LabelCreatedEvent.class));
   }
+
+  @Test
+  public void testRenameNewNameExists() throws Exception {
+    typeContext = new LabelClientTypeContext(store, LabelType.METRIC, metrics, eventBus,
+        MAX_CACHE_SIZE);
+    // Can not rename to a name that is already taken
+    typeContext.createId("newName").get();
+    try {
+      typeContext.rename("oldName", "newName").get();
+      fail("Should have thrown an exception!");
```

Comment:
There is another exception that rename can throw. No need to catch it, if it throws this there is something wrong.